### PR TITLE
Update README.md with if not rules

### PR DESCRIPTION
**Pull request by jtuberville**
Added proposed rules for using unless and avoiding if not

### CODE REVIEW

**Comment by jtuberville** [comment]
@BRIMIL01 @cyphactor @russCloak Thoughts, flames, questions?

**Comment by realitysetsin** [comment]
Yes. Please add a ```.rubocop.yml``` file to this project!

**Comment by RyanHedges** [comment]
So, I think we've been shying away from `unless` because it usually causes confusion when reading the code some time later. Usually this is because the `something` in `unless something` is named poorly and is better read as `if not something`. 

Most of the code I've seen with :exclamation: write it without the space. This makes it harder to miss the bang. Is the actual intention to write it with the space?

```ruby
if ! something

#vs

if !something
```

I do think it's a good rule for the multiple branching. :bowtie:

File: README.md [hunk 1]
```diff
@@ -367,7 +367,49 @@ problem. A known exception to this rule is in a Gems gemspec file.
   # also good - single-line ternary
   some_variable ? do_positive : do_negative
   ```
+  
+* <a name="unless-vs-if-not"></a>
+  Prefer unless over if ! for tests with single branches
```
Comment (cyphactor) [inline README.md]: I would rewrite this as markdown.

Prefer `unless` over `if !` for tests with single branches.

File: README.md [hunk 2]
```diff
@@ -367,7 +367,49 @@ problem. A known exception to this rule is in a Gems gemspec file.
...
+  end
+
+  # good - 
+  unless something
+     do_thing
+  end
+  ```
+  
+* <a name="avoid-if-not"></a>
+  Do not use if ! in tests with multiple branches
```
Comment (cyphactor) [inline README.md]: Write this as markdown.

Do not use `if !` in tests with multiple branches.

**Comment by cyphactor** [comment]
@jtuberville I like it!

File: README.md (hunk 1 above)
Comment (russCloak) [inline README.md]: I would disagree with this rule. I very much prefer `if !true` for all cases *except* the simplest one liners. Examples:

```ruby
# bad - Too much going on, confusing to read (negative negatives).
do_something unless value.foo != "test" && !bar.size.between?(10..15)

# okish - Much less logic and very clear is OK.
do_something unless should_not_do_it

# bad - Negative to negative thought process can be annoying.
# Cannot grok "do something unless is_a_hat? is not true"
# as opposed to: "do something if it is not a hat"
unless stetson.is_a_hat?
  do_something
end

# okish
if !stetson.is_a_hat?
  do_something
end

# good - A method meant for positive evaluation
if stetson.not_a_hat?
  do_something
end
```

**Pull request merged**